addons/mail/models/discuss/discuss_channel.py

```diff
@@ -578,6 +578,10 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
         self._bus_send("discuss.channel/new_message", payload)
         return rdata
 
+    def _notify_mentioned_channel(self, channel_ids=None, **kwargs):
+        # OVERWRITE
+        pass
+
```

Comment:
I personally prefer to put condition in `mail.thread` code to not notify when mentioning channels inside a channel, it sucks to look at different places of the code to see whole logic.